pytorch_lightning/trainer/trainer.py

```diff
@@ -2009,6 +2009,7 @@ def data_parallel_device_ids(self) -> Optional[List[int]]:
 
     @property
     def lightning_module(self) -> "pl.LightningModule":
+        # TODO: this is actually an optional return
```

Comment:
Not planning to resolve this here but I wanted to point out that the annotation is incorrect.